app/controllers/bag_controller.rb

```diff
@@ -2,11 +2,11 @@
 
 class BagController < ApplicationController
   def download
-    send_file("#{Rails.application.config.bag_path}/#{bag_params}.zip", type: 'application/zip', disposition: 'attachment')
+    send_file("#{Rails.application.config.bag_path}/#{bag_params}.#{params[:format]}", type: "application/#{params[:format]}", disposition: 'attachment')
```

Comment:
I'm worried that if you construct the file path using the `params`, is it possible for someone to construct an attack to read arbitrary files on the server?

I recommend something like this:

1. For the #{params[:format]}, make sure it equals `tar` or `zip` before you use it.  If it's any other value, maybe just ignore whatever the user sent, and just assume it's `zip`.

2. For the `bag_params`, maybe check if there is anything in the string that looks like a file path.  For example, if the value is `../../../somefile.txt`, strip everything else off and just use `somefile.txt`?